File: src/types/project.ts
// ---- Project ----
export interface Project {
  id: string;
  title: string;
  genre: string | null;
  summary: string | null;
  target_words: number | null;
  settings_json: string | null;
  cover_path: string | null;
  created_at: string;
  updated_at: string;
}

export interface CreateProject {
  title: string;
  genre?: string | null;
  summary?: string | null;
  target_words?: number | null;
  cover_path?: string | null;
}

export interface UpdateProject {
  title?: string | null;
  genre?: string | null;
  summary?: string | null;
  target_words?: number | null;
  settings_json?: string | null;
  cover_path?: string | null;
}

// ---- Volume ----
export interface Volume {
  id: string;
  project_id: string;
  title: string;
  order_index: number;
  created_at: string;
  updated_at: string;
}

export interface CreateVolume {
  project_id: string;
  title: string;
}

export interface UpdateVolume {
  title?: string | null;
}

File: src-tauri/src/models/project.rs
use serde::{Deserialize, Serialize};
use sqlx::FromRow;

#[derive(Debug, Clone, Serialize, Deserialize, FromRow)]
pub struct Project {
    pub id: String,
    pub title: String,
    pub genre: Option<String>,
    pub summary: Option<String>,
    pub target_words: Option<i64>,
    pub settings_json: Option<String>,
    pub cover_path: Option<String>,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateProject {
    pub title: String,
    pub genre: Option<String>,
    pub summary: Option<String>,
    pub target_words: Option<i64>,
    pub cover_path: Option<String>,
}

#[derive(Debug, Deserialize)]
pub struct UpdateProject {
    pub title: Option<String>,
    pub genre: Option<String>,
    pub summary: Option<String>,
    pub target_words: Option<i64>,
    pub settings_json: Option<String>,
    pub cover_path: Option<String>,
}
